lib/Rx/Operator/ZipOperator.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Rx\Operator;
+
+use Rx\Disposable\CompositeDisposable;
+use Rx\ObservableInterface;
+use Rx\Observer\CallbackObserver;
+use Rx\ObserverInterface;
+use Rx\SchedulerInterface;
+
+class ZipOperator implements OperatorInterface {
+    /** @var ObservableInterface[] */
+    private $sources;
+    
+    /** @var callable */
+    private $resultSelector;
+
+    /** @var \SplQueue[] */
+    private $queues = [];
+
+    /** @var int */
+    private $completesRemaining = 0;
+
+    public function __construct(array $sources, callable $resultSelector = null)
+    {
+        $this->sources = $sources;
+
+        if ($resultSelector === null) {
+            $resultSelector = function () {
+                return func_get_args();
+            };
+        }
+        $this->resultSelector = $resultSelector;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function __invoke(
+        ObservableInterface $observable,
+        ObserverInterface $observer,
+        SchedulerInterface $scheduler = null
+    ) {
+        array_unshift($this->sources, $observable);
+
+        $disposable = new CompositeDisposable();
+
+        $this->completesRemaining = count($this->sources);
+
+        for ($i = 0; $i < count($this->sources); $i++) {
+            $this->queues[$i] = new \SplQueue();
+        }
+
+        for($i=0;$i<count($this->sources);$i++) {
```

Comment:
nit: whitespace `for ($i = 1; ...)`

Also let's introduce a new field in this operator `$this->sourceCount`/`$this->numberOfSources` or the like instead of using `count()` everywhere.